pkg/codegen/hcl2/binder_schema.go

```diff
@@ -124,22 +124,26 @@ func (b *binder) loadPackageSchema(name string) error {
 }
 
 // schemaTypeToType converts a schema.Type to a model Type.
-func schemaTypeToType(src schema.Type) model.Type {
+func (b *binder) schemaTypeToType(src schema.Type) (result model.Type) {
+	defer func() {
+		b.typeSchemas[result] = src
```

Comment:
This is an interesting approach. I guess the inline alternative would be some form of goto?

Reply:
Correct.